lnrpc/routerrpc/router_server_test.go

```diff
@@ -214,3 +217,177 @@ func TestTrackPaymentsNoInflightUpdates(t *testing.T) {
 	payment := <-stream.sentFromServer
 	require.Equal(t, lnrpc.Payment_SUCCEEDED, payment.Status)
 }
+
+func TestIsLsp(t *testing.T) {
+	alicePubKey, err := btcec.ParsePubKey([]byte{
+		0x2,
+		0xea, 0x41, 0x48, 0x11, 0x2, 0x59, 0xe3, 0x5c,
+		0x51, 0x15, 0x90, 0x25, 0x4a, 0x61, 0x5, 0x51,
+		0xb3, 0x8, 0xe9, 0xd5, 0xf, 0xc6, 0x91, 0x25,
+		0x14, 0xd2, 0xcf, 0xc8, 0xc5, 0x5b, 0xd9, 0x88,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	bobPubKey, err := btcec.ParsePubKey([]byte{
+		0x2,
+		0x6, 0x16, 0xd1, 0xb1, 0x4f, 0xee, 0x11, 0x86,
+		0x55, 0xfe, 0x31, 0x66, 0x6f, 0x43, 0x1, 0x80,
+		0xa8, 0xa7, 0x5c, 0x2, 0x92, 0xe5, 0x7c, 0x4,
+		0x31, 0xa6, 0xcf, 0x43, 0xb6, 0xdb, 0xe6, 0x10,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	carolPubKey, err := btcec.ParsePubKey([]byte{
+		0x2,
+		0x88, 0x65, 0x16, 0xc2, 0x37, 0x3f, 0xc5, 0x16,
+		0x62, 0x71, 0x0, 0xdd, 0x4d, 0x43, 0x28, 0x43,
+		0x32, 0x91, 0x75, 0xcc, 0xd8, 0x81, 0xb6, 0xb0,
+		0xd8, 0x96, 0x78, 0xad, 0x18, 0x3b, 0x16, 0xe1,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	davePubKey, err := btcec.ParsePubKey([]byte{
+		0x3,
+		0xfa, 0x1f, 0x6, 0x3a, 0xa4, 0x75, 0x2e, 0x74,
+		0x3e, 0x55, 0x9, 0x20, 0x6e, 0xf6, 0xa8, 0xe1,
+		0xd7, 0x61, 0x50, 0x75, 0xa8, 0x34, 0x15, 0xc3,
+		0x6b, 0xdc, 0xb0, 0xbf, 0xaa, 0x66, 0xd7, 0xa7,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var (
+		aliceHopHint = zpay32.HopHint{
+			NodeID:                    alicePubKey,
+			FeeBaseMSat:               100,
+			FeeProportionalMillionths: 1_000,
+			ChannelID:                 421337,
+		}
+
+		bobHopHint = zpay32.HopHint{
+			NodeID:                    bobPubKey,
+			FeeBaseMSat:               2_000,
+			FeeProportionalMillionths: 2_000,
+			ChannelID:                 815,
+		}
+
+		carolHopHint = zpay32.HopHint{
+			NodeID:                    carolPubKey,
+			FeeBaseMSat:               2_000,
+			FeeProportionalMillionths: 2_000,
+			ChannelID:                 815,
+		}
+
+		daveHopHint = zpay32.HopHint{
+			NodeID:                    davePubKey,
+			FeeBaseMSat:               2_000,
+			FeeProportionalMillionths: 2_000,
+			ChannelID:                 815,
+		}
+
+		emptyRouteHints = [][]zpay32.HopHint{}
+	)
+
+	// Empty route hints should not be considered an LSP.
+	isLsp, adjustedRouteHints, lspHint := isLSP([][]zpay32.HopHint{
+		{},
+	})
+	require.False(t, isLsp)
+	require.True(t, reflect.DeepEqual(emptyRouteHints, adjustedRouteHints))
+	require.Nil(t, lspHint)
+
+	// A single hop should not be considered an LSP.
```

Comment:
this comment does not match the test case below it

Reply:
"should be"